test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3531 @@
+        // formula:
+        // (depositAmount * totalSupply) / total
+        // (1 * 1) / 2 = 0.5 shares
+        await this.vault.connect(counterpartySigner).deposit(depositAmount);
+        assert.equal(
+          (await this.vault.balanceOf(counterparty)).toString(),
+          BigNumber.from("50000000000")
+        );
+      });
+
+      it("reverts when no value passed", async function () {
+        await expect(
+          this.vault.connect(userSigner).deposit(0)
+        ).to.be.revertedWith(/Insufficient asset balance/);
+      });
+
+      it("does not inflate the share tokens on initialization", async function () {
+        const depositAmount = BigNumber.from("100000000000");
+
+        await this.assetContract
+          .connect(adminSigner)
+          .transfer(this.vault.address, depositAmount);
+
+        await this.vault
+          .connect(userSigner)
+          .deposit(BigNumber.from("10000000000"));
+
+        // user needs to get back exactly 1 ether
+        // even though the total has been incremented
+        assert.isFalse((await this.vault.balanceOf(user)).isZero());
+      });
+
+      it("reverts when minimum shares are not minted", async function () {
+        await expect(
+          this.vault
+            .connect(userSigner)
+            .deposit(
+              BigNumber.from(this.minimumSupply).sub(BigNumber.from("1"))
+            )
+        ).to.be.revertedWith(/Insufficient asset balance/);
+      });
+    });
+
+    describe("#depositYieldToken", () => {
+      time.revertToSnapshotAfterEach();
+
+      beforeEach(async function () {
+        const addressToDeposit = [
+          userSigner,
+          managerSigner,
+          counterpartySigner,
+          adminSigner,
+        ];
+
+        // Deposit only if asset is WETH
+        if (params.depositAsset === WETH_ADDRESS) {
+          for (let i = 0; i < addressToDeposit.length; i++) {
+            const weth = this.assetContract.connect(addressToDeposit[i]);
+            await weth.deposit({ value: parseEther("10") });
+            await weth.approve(this.vault.address, parseEther("3"));
+          }
+        }
+
+        await setupYieldToken(
+          addressToDeposit,
+          this.assetContract,
+          this.collateralAsset,
+          this.vault,
+          parseEther("7")
+        );
+      });
+
+      it("deposits successfully", async function () {
+        const depositAmount = BigNumber.from("100000000000");
+        let pricePerShare = BigNumber.from(
+          (await this.collateralContract.pricePerShare()).toString()
+        );
+        const depositAmountInAsset = wmul(depositAmount, pricePerShare);
+
+        await this.assetContract
+          .connect(userSigner)
+          .approve(this.vault.address, depositAmount);
+
+        const res = await this.vault.depositYieldToken(depositAmount);
+        const receipt = await res.wait();
+        assert.isAtMost(receipt.gasUsed.toNumber(), 170000);
```

Comment:
Check for `totalBalance` and `assetBalance` here too

Reply:
👍